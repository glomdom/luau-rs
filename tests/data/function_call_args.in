fn main(a: i32) -> i32 {
    test_fn(a)
}

fn test_fn(a: i32) -> i32 {
    a + 2
}