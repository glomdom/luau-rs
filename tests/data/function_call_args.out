Block(
    Block {
        statements: [
            Function(
                Function {
                    name: "main",
                    params: [
                        Param {
                            name: "a",
                            typ: LuauType {
                                type_name: "number",
                                is_mut: false,
                                is_ref: false,
                            },
                        },
                    ],
                    ret_type: Some(
                        LuauType {
                            type_name: "number",
                            is_mut: false,
                            is_ref: false,
                        },
                    ),
                    body: Block(
                        Block {
                            statements: [
                                Return(
                                    Return {
                                        value: Some(
                                            Call(
                                                Call {
                                                    func: "test_fn",
                                                    args: [
                                                        Value(
                                                            Value {
                                                                value: "a",
                                                            },
                                                        ),
                                                    ],
                                                },
                                            ),
                                        ),
                                    },
                                ),
                            ],
                        },
                    ),
                },
            ),
            Function(
                Function {
                    name: "test_fn",
                    params: [
                        Param {
                            name: "a",
                            typ: LuauType {
                                type_name: "number",
                                is_mut: false,
                                is_ref: false,
                            },
                        },
                    ],
                    ret_type: Some(
                        LuauType {
                            type_name: "number",
                            is_mut: false,
                            is_ref: false,
                        },
                    ),
                    body: Block(
                        Block {
                            statements: [
                                Return(
                                    Return {
                                        value: Some(
                                            BinaryOp(
                                                BinaryOp {
                                                    op: "+",
                                                    left: Value(
                                                        Value {
                                                            value: "a",
                                                        },
                                                    ),
                                                    right: Value(
                                                        Value {
                                                            value: "2",
                                                        },
                                                    ),
                                                },
                                            ),
                                        ),
                                    },
                                ),
                            ],
                        },
                    ),
                },
            ),
        ],
    },
)